### ISSUE TITLE
black screen

### ISSUE THREAD

**Issue by protrading104** [PhrozenIO/Arcane#2]
I see a black screen when connecting to win serv 2012. can this be solved ?

![image](https://github.com/user-attachments/assets/2965eaef-1be3-4906-a3c6-57a731c097fa)

**log**

└─$ arcane-viewer
Ignoring window icon 4194306 exceeds maximum xcb request length 4194303
Ignoring window icon 4194306 exceeds maximum xcb request length 4194303
2024-08-06 13:49:25,609 - arcane_viewer.arcane.client - INFO - Connecting to remote server: `10.0.0.10:2801`...
2024-08-06 13:49:25,610 - arcane_viewer.arcane.client - DEBUG - Creating and configure new SSL context...
Ignoring window icon 4194306 exceeds maximum xcb request length 4194303
2024-08-06 13:49:25,703 - arcane_viewer.arcane.client - DEBUG - Establishing connection to remote server...
Ignoring window icon 4194306 exceeds maximum xcb request length 4194303
2024-08-06 13:49:25,722 - arcane_viewer.arcane.client - INFO - [18] Connected! Authenticating with remote server...
2024-08-06 13:49:25,722 - arcane_viewer.arcane.client - DEBUG - Request challenge...
2024-08-06 13:49:25,774 - arcane_viewer.arcane.client - DEBUG - Received challenge: `FF1D1F5EA570C5D2830A5B2A3CBD2BFA43CDC76E69A0F45521FF6AD6F14CD714437700E0396BF6C64B5C4AD45297A25205351E8916549DB6121C27FF99BBE465`, attempt to solve it with defined password...
2024-08-06 13:49:25,775 - arcane_viewer.arcane.client - DEBUG - Challenge solved: `C4A4ABDCDC1A95DA9876E1F19EA3EC36B087ADBA6C9FD6FF7C161E779EA56EB97D9A51EB9851727D0B7DD37C7DDDE7B890DBFC871A6846CACC1EB654BF6B11A7`, sending solution to server...
2024-08-06 13:49:25,777 - arcane_viewer.arcane.client - INFO - Authentication successful
2024-08-06 13:49:25,827 - arcane_viewer.arcane.session - DEBUG - @Session information:
2024-08-06 13:49:25,827 - arcane_viewer.arcane.session - DEBUG - {
    "Username": "SYSTEM",
    "Version": "5.0.0b1",
    "ViewOnly": false,
    "MachineName": "DC1",
    "WindowsVersion": "Microsoft Windows NT 6.3.9600.0",
    "SessionId": "2D41908F226A4494FFB241163F4620DF105DC077C3BA84D2BA2BEB39211FE1E962F3A6EA46334B9495F7AE24F1DECDF670885ADBB8433A5C26D254A3C0CA5228"
}
2024-08-06 13:49:25,827 - arcane_viewer.arcane.session - INFO - Session established with remote peer
2024-08-06 13:49:25,827 - arcane_viewer.arcane.client - INFO - [18] Closing connection...
Ignoring window icon 4194306 exceeds maximum xcb request length 4194303
Ignoring window icon 4194306 exceeds maximum xcb request length 4194303
2024-08-06 13:49:25,860 - arcane_viewer.arcane.client - INFO - Connecting to remote server: `10.0.0.10:2801`...
2024-08-06 13:49:25,862 - arcane_viewer.arcane.client - DEBUG - Creating and configure new SSL context...
2024-08-06 13:49:25,874 - arcane_viewer.arcane.client - DEBUG - Establishing connection to remote server...
2024-08-06 13:49:25,884 - arcane_viewer.arcane.client - INFO - [18] Connected! Authenticating with remote server...
2024-08-06 13:49:25,884 - arcane_viewer.arcane.client - DEBUG - Request challenge...
2024-08-06 13:49:25,936 - arcane_viewer.arcane.client - DEBUG - Received challenge: `F4C9033672DE966A3590DFD579142A1BA52D5C2E2D61F64E9A91ACF5FC12B9ED9A9B0CC0B18E3C104C7ED46B19820FC654280178218DB75F6E2D6DFF32DEB44C`, attempt to solve it with defined password...
2024-08-06 13:49:25,938 - arcane_viewer.arcane.client - DEBUG - Challenge solved: `59C12DAD6629A421D65A1B72492EE8049B7D05D80BCF524AD68A8C6918FA60CF65DC2F9BBC7BF00BD82648C53B4096B85A51DB8808DB5F5D3666AEFD86D35D7A`, sending solution to server...
2024-08-06 13:49:25,940 - arcane_viewer.arcane.client - INFO - Authentication successful
2024-08-06 13:49:26,288 - arcane_viewer.arcane.threads.v_desktop - INFO - 1 screen(s) detected
2024-08-06 13:49:26,289 - arcane_viewer.arcane.threads.v_desktop - INFO - Screen: WinDisc (1024x768)
2024-08-06 13:49:26,296 - arcane_viewer.arcane.client - INFO - Connecting to remote server: `10.0.0.10:2801`...
2024-08-06 13:49:26,297 - arcane_viewer.arcane.client - DEBUG - Creating and configure new SSL context...
2024-08-06 13:49:26,315 - arcane_viewer.arcane.client - DEBUG - Establishing connection to remote server...
2024-08-06 13:49:26,339 - arcane_viewer.arcane.client - INFO - [24] Connected! Authenticating with remote server...
2024-08-06 13:49:26,339 - arcane_viewer.arcane.client - DEBUG - Request challenge...
2024-08-06 13:49:26,398 - arcane_viewer.arcane.client - DEBUG - Received challenge: `4BC07011F94ED11EB81BAE8223CCEB49312E427AA06AFFB231D89F5BCFF93872BA3C6871DE9164B4D7660188FE0888C97DF8DC121B1D70EBD9703CB5AC13167A`, attempt to solve it with defined password...
2024-08-06 13:49:26,400 - arcane_viewer.arcane.client - DEBUG - Challenge solved: `EDEED396598C22DAB4BDFF4B83C8A6F87F3D6EF84387100E63730873021A0D461D85522D8F1BBCA50879F81BD923390E8A31D893E9B5BF43B7C36453D2177035`, sending solution to server...
2024-08-06 13:49:26,402 - arcane_viewer.arcane.client - INFO - Authentication successful
^C

**settings**
![image](https://github.com/user-attachments/assets/b1dee6c3-ed8c-4e7e-b1f3-7107b4fc9637)

**on the  Server:**
$certPassword = ConvertTo-SecureString -String 'yourPassword' -AsPlainText -Force

Invoke-ArcaneServer -ListenAddress '0.0.0.0' -ListenPort 2801 -SecurePassword (ConvertTo-SecureString -String 'urCompl3xP@ssw0rd!@#' -AsPlainText -Force) -CertificateFile 'C:\\ProgramData\\cert.pfx' -CertificatePassword $certPassword"

**Comment by DarkCoderSc** [comment]
Is the server connected to an external display or a HDMI dummy plug?

If not, and if you're unable to connect one, could you try using Arcane while actively connected through Windows RDP?

Thank you in advance.

**Comment by protrading104** [comment]
10.0.0.10 (win serv 2012) - this is my virtual machine
 no external displays

**Comment by DarkCoderSc** [comment]
Have you tried while being actively connected with RDP ?

**Comment by protrading104** [comment]
no. does it have to be RDP active ?  I thought that was an alternative.

**Comment by DarkCoderSc** [comment]
To function correctly, Arcane requires the target computer to have one of the following:

- A Physical Display
- An HDMI Dummy Plug (simulate a display device).
- A Virtual Display (Open-source drivers are available to set up such a virtual display; I will cover the setup process in a future post)
- An Active Remote Desktop Connection that uses a virtual display (RDP, VNC).

In short, Arcane relies on capturing and transmitting the visual output from an active desktop using Windows API, BitBlt, which is a GDI (Graphics Device Interface) function. BitBlt operates on the device context of the display, copying pixel data from the screen for remote rendering.

However, if the computer does not have any active display, whether physical or virtual, there is no device context available for Arcane to capture and BitBlt cannot function.

Arcane is a unique alternative to traditional remote desktop software, though it is more limited in scope. This limitation stems from the fact that the server component is written entirely in PowerShell. I have no intention of implementing the server or any other part of the server in another language, as the primary goal and true purpose of this project is to create a remote desktop solution fully developed in PowerShell.

In the future, Arcane will feature many enhancements that will distinguish it from other remote desktop applications. However, regarding the current display requirement, the best workaround in your situation would be to set up a virtual display using one of the many open-source drivers available.

I plan to write a guide on how to set up a virtual display, so stay tuned for that! 😉

**Comment by protrading104** [comment]
ok, 
thanks for explaining
